tests/end_to_end/features/steps/autovariants.py

```diff
@@ -0,0 +1,132 @@
+import os
+import random
+
+import featureform as ff
+from behave import when, then
+
+
+@when("I turn on autovariants")
+def step_impl(context):
+    os.environ["FF_GET_EQUIVALENT_VARIANTS"] = "true"
+
+    context.variant = ff.get_run()
+
+
+@then("I turn off autovariants")
+def step_impl(context):
+    os.environ["FF_GET_EQUIVALENT_VARIANTS"] = "false"
+
+    context.variant = ff.get_run()
+
+
+@when("I register a transformation with auto variant")
+def step_impl(context):
```

Comment:
Ideally this is "I register a transformation with {{variant}}"

And if it's empty we default to autovariants.